src/app/app.module.ts

```diff
@@ -1,44 +1,27 @@
-import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
-import { HttpClientModule} from '@angular/common/http'
+import { ShopModule } from './modules/shop.module';
+import { SharedModule } from './modules/shared.module';
 
-import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
 import { HeaderComponent } from './components/header/header.component';
-import { CartComponent } from './components/header/cart/cart.component';
-import { CartItemsComponent } from './components/header/cart/cart-items/cart-items.component';
-import { FooterComponent } from './components/footer/footer.component';
-import { ShoppingComponent } from './components/shopping/shopping.component';
-import { ProductsComponent } from './components/shopping/products/products.component';
-import { ProductItemsComponent } from './components/shopping/products/product-items/product-items.component';
 import { BreadcrumbComponent } from './components/breadcrumb/breadcrumb.component';
-import { SpinnerComponent } from './components/spinner/spinner.component';
-import { ProductDetailsComponent } from './components/shopping/products/product-details/product-details.component';
-import { ViewCartComponent } from './components/shopping/view-cart/view-cart.component';
 import { HomeComponent } from './components/home/home.component';
 import { NotFoundComponent } from './components/not-found/not-found.component';
+import { FooterComponent } from './components/footer/footer.component';
+
 
 @NgModule({
   declarations: [
     AppComponent,
     HeaderComponent,
-    CartComponent,
-    CartItemsComponent,
-    FooterComponent,
-    ShoppingComponent,
-    ProductsComponent,
-    ProductItemsComponent,
     BreadcrumbComponent,
-    SpinnerComponent,
-    ProductDetailsComponent,
-    ViewCartComponent,
     HomeComponent,
-    NotFoundComponent
+    NotFoundComponent,
+    FooterComponent
   ],
   imports: [
-    BrowserModule,
-    AppRoutingModule,
-    HttpClientModule
+    ShopModule,
+    SharedModule,
```

Comment:
keep below modules.

```
GuestModule
Listingmodule
DetailModule
CartModule
```

Reply:
Listingmodule - will it contain all lists like Products, Users, orders ... etc? and similarly DetailModule?

Why Shop module need to be split further?

Reply:
Updated as per the comment